Nanocloud/community
# Allow to test Nanocloud's API with mocha

Implement API testing in Javascript using mocha test framework and chai with some customization for json-apis and Nanocloud.
Introduce nanotest to facilitate API tests writting in javascript for Nanocloud.

So far, nanotest is able to : 
- Login with multiple user at the same time
- Perform authenticated request
- Perform post, get and delete requests
- Set custom header value
- Check HTTP status
- Validate json-api structure
- Schema returned json against schemas
- Check headers values
- \+ all chai assertions

It currently tests login (both valid and invalid attempt), creating users and removing users.
- Not all tests currently pass since Nanocloud is not yet a strictly valid json-api *
  This PR aims to provide the first block toward a true alternative to postman tests which are not easy to maintain, error-prone and not git diff friendly.
